src/sparsezoo/model/model.py

```diff
@@ -15,52 +15,72 @@
 import logging
 import os
 import re
-from typing import Any, Dict, Generator, List, Optional, Union
+from typing import Any, Dict, Generator, List, Optional, Tuple, Union
 
 import numpy
 
-from sparsezoo.v2.inference.engines import ENGINES
-from sparsezoo.v2.inference.inference_runner import InferenceRunner
-from sparsezoo.v2.objects.directory import Directory, is_directory
-from sparsezoo.v2.objects.file import File
-from sparsezoo.v2.objects.model_objects import NumpyDirectory
-
-
-__all__ = ["ModelDirectory"]
-
-
-def file_dictionary(**kwargs):
-    return kwargs
-
-
-class ModelDirectory(Directory):
+from sparsezoo.inference import ENGINES, InferenceRunner
+from sparsezoo.model.utils import (
+    CACHE_DIR,
+    ZOO_STUB_PREFIX,
+    load_files_from_directory,
+    load_files_from_stub,
+    save_outputs_to_tar,
+)
+from sparsezoo.objects import (
+    Directory,
+    File,
+    NumpyDirectory,
+    SelectDirectory,
+    is_directory,
+)
+from sparsezoo.validation import IntegrationValidator
+
+
+__all__ = ["Model"]
+
+ALLOWED_CHECKPOINT_VALUES = {"prepruning", "postpruning", "preqat", "postqat"}
+ALLOWED_RECIPE_VALUES = {"original", "transfer_learn"}
+ALLOWED_DEPLOYMENT_VALUES = {"default"}
+
+PARAM_DICT = {
+    "checkpoint": ALLOWED_CHECKPOINT_VALUES,
+    "recipe": ALLOWED_RECIPE_VALUES,
+    "deployment": ALLOWED_DEPLOYMENT_VALUES,
+}
+
+
+class Model(Directory):
     """
     Object to represent SparseZoo Model Directory.
 
-    The suggested way to create a class object is to use
-    any of the two factory methods:
+    :param path: a string argument that can be one of two things:
+        a) a SparseZoo model stub:
+            i) without additional string arguments
+                e.g. 'zoo:model/stub/path'
+            ii) with additional string arguments (that will restrict
+                the set of file/directory files handled by this
+                `Model` object)
+                e.g. 'zoo:model/stub/path?param1=value1&param2=value2'
+        b) a local directory path
+            e.g. `/home/user/model_path`
+    """
```

Comment:
This class is super easy to use from python interpreter. Maybe just add something to docstring about which methods/propeties should be used? Seems like generate_outputs() and maybe path are main ones?